From 745d3863464ce8b2124419b677fdf4b48ffafafe Mon Sep 17 00:00:00 2001
From: SongFerngWang <songferngwang@google.com>
Date: Wed, 5 May 2021 21:30:39 +0800
Subject: [PATCH] SubscriptionGroup is exposed to unprivileged callers

SubscriptionInfo.mGroupUUID is not cleared in
conditionallyRemoveIdentifiers if the caller only has READ_PHONE_STATE
(based on a check to checkReadPhoneState) and not READ_DEVICE_IDENTIFIERS.
Bug: 181053462
Test: atest SubscriptionManagerTest

Change-Id: I68d1edb4e7cc2ad6696363ea1dacb09e839a651e
Merged-In: I68d1edb4e7cc2ad6696363ea1dacb09e839a651e
(cherry picked from commit 1399361f2e553a673b4a0d90131c44ee2fe0e32a)
---
 .../com/android/internal/telephony/SubscriptionController.java   | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/java/com/android/internal/telephony/SubscriptionController.java b/src/java/com/android/internal/telephony/SubscriptionController.java
index bae9a700c..8a03431f9 100644
--- a/src/java/com/android/internal/telephony/SubscriptionController.java
+++ b/src/java/com/android/internal/telephony/SubscriptionController.java
@@ -4085,6 +4085,7 @@ public class SubscriptionController extends ISub.Stub {
         if (!hasIdentifierAccess) {
             result.clearIccId();
             result.clearCardString();
+            result.clearGroupUuid();
         }
         if (!hasPhoneNumberAccess) {
             result.clearNumber();
-- 
2.17.1

